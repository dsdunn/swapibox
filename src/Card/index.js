import React from 'react';

const Card = ({person, vehicle, planet, toggleFavorites}) => {

const displayResidents = (residents) => {
  return residents.map(resident => <li key={resident}>{resident}</li> )
}

  const Card = person ? 
        <div className="card"> 
          <button onClick={toggleFavorites} name={person.name}> Favorite </button>
          <h2>name: {person.name}</h2>
          <p>species: {person.species}</p>
          <p>Homeworld: {person.planetName}</p>
          <p>population: {person.population}</p>
        </div>
        : vehicle ?
        <div>
          <button onClick={toggleFavorites}> Favorite </button>
          <h2>name: {vehicle.name}</h2>
          <p>model: {vehicle.model}</p>
          <p>class: {vehicle.vehicle_class}</p>
          <p>Passengers: {vehicle.passengers}</p>
        </div>   
        : 
        <div>
          <button onClick={toggleFavorites}> Favorite </button>
          <h2>name: {planet.name}</h2>
          <p>population: {planet.population}</p>
          <p>climate: {planet.climate}</p>
          <ul>residents: {displayResidents(planet.residents)}</ul>
        </div>
  return Card;
}

export default Card;