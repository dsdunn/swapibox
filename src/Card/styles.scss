
.card {
  margin: 3%;
  border: 1px solid black;
  padding: 2%;
  width: 25vw;
  background-color: #d8d8d2;
  font-family: Sans-Serif;
}

.favorite {
  border: 2px solid red;
}

.card p {
  font-size: 1.1em;
  margin: auto;
}

.nofav {
  color: gray;
}