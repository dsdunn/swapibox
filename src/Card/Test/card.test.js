import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Card from '../'

describe('Card', () => {
  let wrapper;
  let mockToggleFavs = jest.fn();
  let mockEvent

  beforeEach(() => {
        wrapper = shallow(<Card 
                        individual={{species: 'human'}}
                        toggleFavorites={mockToggleFavs}
                        isFavorite={true}
                      />)
  })

  it('matches the snapShot', () => {


    expect(wrapper).toMatchSnapshot();
  })

  it('renders the correct card for the type of individual', () => {
    expect(wrapper.find('.person').length).toEqual(1);
    expect(wrapper.find('.vehicle').length).toEqual(0);
    expect(wrapper.find('.favorite').length).toEqual(1);
  })

  it('should call toggleFavorites on click of button', () => {
    wrapper.find('button').simulate('click')
    expect(mockToggleFavs).toHaveBeenCalled();
  })

  it('should call displayResidents if individual is a planet', () => {
    wrapper = shallow(<Card 
                    individual={{residents: ['one','two']}}
                    toggleFavorites={mockToggleFavs}
                    isFavorite={true}
                  />)

    expect(wrapper.find('li').length).toEqual(2)
  })

  it('should apply className of favorite if isFavorite is true', () => {
    wrapper = shallow(<Card 
                    individual={{residents: ['one','two']}}
                    toggleFavorites={mockToggleFavs}
                    isFavorite={false}
                  />)
    expect(wrapper.find('.favorite').length).toEqual(0)
  })

  it('should render the appropriate class name on the card', () => {
        wrapper = shallow(<Card 
                    individual={{vehicle_class: "clunker"}}
                    toggleFavorites={mockToggleFavs}
                    isFavorite={false}
                  />)
        expect(wrapper.find('.vehicle').length).toEqual(1);
        expect(wrapper.find('.favorite').length).toEqual(0);
  })
})