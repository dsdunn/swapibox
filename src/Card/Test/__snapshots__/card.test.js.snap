// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    individual={
        Object {
            "species": "human",
          }
    }
    isFavorite={true}
    toggleFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          name={undefined}
          onClick={[Function]}
>
           Favorite 
</button>,
        <h3 />,
        <p>
          species: 
          human
</p>,
        <p>
          Homeworld: 
</p>,
        <p>
          population: 
</p>,
      ],
      "className": "card person favorite",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Favorite ",
          "name": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": " Favorite ",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "species: ",
            "human",
          ],
        },
        "ref": null,
        "rendered": Array [
          "species: ",
          "human",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Homeworld: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Homeworld: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "population: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "population: ",
          undefined,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            name={undefined}
            onClick={[Function]}
>
             Favorite 
</button>,
          <h3 />,
          <p>
            species: 
            human
</p>,
          <p>
            Homeworld: 
</p>,
          <p>
            population: 
</p>,
        ],
        "className": "card person favorite",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Favorite ",
            "name": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": " Favorite ",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "species: ",
              "human",
            ],
          },
          "ref": null,
          "rendered": Array [
            "species: ",
            "human",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Homeworld: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Homeworld: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "population: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "population: ",
            undefined,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
